### DISCUSSION TITLE
Official pressure drop bench support thread

### DISCUSSION THREAD

**Discussion by DeeEmm** [DIY-Flow-Bench/DIY-Flow-Bench#173] [Code Development]
Item #27 on the roadmap - Support for pressure drop benches

This was originally scheduled for the V3 release but I am bringing it forwards@ request of @danielerpl

There are a number of ways to go about this, but it make sense to develop the feature with the help of someone running / building a pressure drop style bench so that functionality can be validated as we go along.

Intent is to develop for orifice style bench and then apply to pitot & venturi. Most of the code will be directly applicable to all.

### Considerations

- Need to be able to store flow rates for orifices.
- Need to be able to select which orifice is in use
- Ideally need to be able to automatically detect which orifice is being used

Basic bench style determination is carried out in DIY-Flow-Bench.ino I have modded the current code to create the necessary decision pathways...

```
        // Get flow data
        // Bench is MAF type...
        if (strstr(String(config.bench_type).c_str(), String("MAF").c_str())){
          #ifdef MAF_IS_ENABLED
          sensorVal.FlowKGH = _sensors.getMafFlow();
          sensorVal.FlowCFM = _calculations.convertFlow(sensorVal.FlowKGH);
          // sensorVal.FlowCFM = _calculations.convertMassFlowToVolumetric(sensorVal.FlowKGH);
          // sensorVal.FlowCFM = _calculations.convertKGHtoCFM(sensorVal.FlowKGH) + calVal.flow_offset;
          #endif

        // Bench is ORIFICE type...
        } else if (strstr(String(config.bench_type).c_str(), String("ORIFICE").c_str())){


        // Bench is VENTURI type...
        } else if (strstr(String(config.bench_type).c_str(), String("VENTURI").c_str())){


        // Bench is PITOT type...
        } else if (strstr(String(config.bench_type).c_str(), String("PITOT").c_str())){


        // Error bench type unknown
        } else {


        }

```

Need to create method for calculating flow from differential pressure.

This first requires orifice flow data. We already have facility to store orifice data in configuration...

<img width="787" alt="Screenshot 2024-09-21 at 8 21 34 PM" src="https://github.com/user-attachments/assets/00266aab-1114-4cb1-985d-dd3aa744def7">

And ability to select current orifice in use...

<img width="1350" alt="Screenshot 2024-09-21 at 8 23 02 PM" src="https://github.com/user-attachments/assets/683081a7-acc2-40c3-8d2a-a76a56b01a4c">

But as you can see there are some errors to address.

The next step after fixing the errors is to create the method which takes differential pressure and calculates flow from the orifice flow data.

**Comment by DeeEmm** [comment]
Progress update!

I've resolved the error above. We can now select active orifice in the GUI.

Active orifice calibrated flow rate along with pRef and pDiff are used to calculate flow rate using ratiometric principles. i.e. percentage of calibrated flow rate.

This is the differential flow method that I created

```


/***********************************************************
 * @brief Returns flow value in CFM from differential pressure using ratiometric principle
 * @note uses calibrated orifice flow rate and assumes reference pressure is matched to orifice calibration
 * @note pRef tapped below orifice plate | pDiff tapped above orifice plate
 *
 ***/
double Sensors::getDifferentialFlow() {

	extern struct DeviceStatus status;
	extern struct SensorData sensorVal;
	extern struct ConfigSettings config;
	extern struct Translator translate;

	Hardware _hardware;
    Messages _message;  

	double flowRateCFM = 0.0;

    double orificeOneFlow = 0.0;
    double orificeOneDepression = 0.0;

	double orificeFlowRate = 0.0;
	double orificeDepression = 0.0;
	double flowRatio = 0.0;

	// get orifice flow data from active orifice
	switch (status.activeOrifice)   {

		case '1': 
			orificeFlowRate = config.orificeOneFlow;
			orificeDepression = config.orificeOneDepression;
		break;

		case '2': 
			orificeFlowRate = config.orificeTwoFlow;
			orificeDepression = config.orificeTwoDepression;
		break;

		case '3': 
			orificeFlowRate = config.orificeThreeFlow;
			orificeDepression = config.orificeThreeDepression;
		break;

		case '4': 
			orificeFlowRate = config.orificeFourFlow;
			orificeDepression = config.orificeFourDepression;
		break;

		case '5': 
			orificeFlowRate = config.orificeFiveFlow;
			orificeDepression = config.orificeFiveDepression;
		break;

		case '6': 
			orificeFlowRate = config.orificeSixFlow;
			orificeDepression = config.orificeSixDepression;
		break;

		// We've got here without a valid active orifice so lets set something and send an error message
        default:
			orificeFlowRate = config.orificeOneFlow;
			orificeDepression = config.orificeOneDepression;
			_message.serialPrintf("%s\n", "Invalid Orifice Data");
			String statusMessage = translate.LANG_VAL_INVALID_ORIFICE_SELECTED;
        break;

	}

	// Calculate flow rate based on calibrated orifice data, ref pressure and current pressure drop
	flowRatio = sensorVal.PDiffKPA / sensorVal.PRefKPA;
	flowRateCFM = flowRatio * orificeFlowRate;

	return flowRateCFM;

}
```

There are some caveats at present.

- This is untested
- The reference pressure must be the same that the orifice was calibrated at

I will package this up and update the repo.

Now I just need someone to test.

**Comment by DeeEmm** [comment]
OK. So changes have now been merged with master (Build 24092301)

To use bench as differential pressure / orifice style bench...

### Configuration

- Enable orifice bench support under config settings tab in GUI. Save.
- Enter orifice flow data. Save

You will now see the orifice selector on the main GUI screen

- Select orifice. Orifice calibration data will be shown below orifice selector.

### Hardware setup

pRef sensor needs to be connected below the orifice plate
pDiff sensor needs to be connected between orifice plate and test piece

### Test Depression

At the moment the code will only return accurate results if the test depression (pRef) matches that which the orifice was calibrated at.

I will mod the code to offset the result so that it is accurate no matter what pRef reads - basically the same as using a lookup table to transpose the results on a traditional bench. 

The ideal scenario here is to also take environmental conditions into consideration. That is the temp / humidity and baro that the orifice was calibrated at. Only then can an orifice bench be truly accurate. Otherwise there will always be an inherent error in the calcs.





Please report back any findings in this thread

**Reply by danielerpl** [reply]
Lovely @DeeEmm, I will make a test for you in the next days. I will give you all information is returned from the board! 
Many thanks for this great and quickly job. I will keep you updated about it as soon as possible

**Reply by DeeEmm** [reply]
Cool. Thanks.

Let me know if you have any issues.

**Reply by danielerpl** [reply]
<img width="1671" alt="Screenshot 2024-09-24 at 14 47 25" src="https://github.com/user-attachments/assets/f247dffa-597e-4efa-a8c4-8d056ffd53cb">
<img width="1663" alt="Screenshot 2024-09-24 at 14 47 33" src="https://github.com/user-attachments/assets/9f1eb587-1d25-4d54-a51c-2e13459b3ce8">

![IMG_7909](https://github.com/user-attachments/assets/b9412487-7df2-4d26-a74a-de53516e1e8a)

So, this is the test I. made with the orifice configured, CFM 100 @28  :  

In the first and second screen you can see the value of Diff Pressure at 0.98 and the flow rate at 11.00
When I tour on all the flow bench motors, at the minimum speed, I have the others value 

<img width="1675" alt="Screenshot 2024-09-24 at 14 47 18" src="https://github.com/user-attachments/assets/9015e55a-7e87-46c5-b167-2984deaf8ffe">
<img width="1674" alt="Screenshot 2024-09-24 at 14 47 07" src="https://github.com/user-attachments/assets/3693e3c8-d677-40ab-bba3-80eddb602244">

Depression 35.75 and CFM 11.01

About the CFM I can see the differente base on the tested piece and base on the amount of valves opened, and this I think it's a very good point ad you can see from a test chart that I made. 

this is the json of te sample cart I made 

{
  "LIFTDATA1": 333.608382,
  "LIFTDATA2": 127.1512918,
  "LIFTDATA3": 0,
  "LIFTDATA4": 53.45046351,
  "LIFTDATA5": 0,
  "LIFTDATA6": 26.4058886,
  "LIFTDATA7": 18.6719361,
  "LIFTDATA8": 0,
  "LIFTDATA9": 0,
  "LIFTDATA10": 0,
  "LIFTDATA11": 0,
  "LIFTDATA12": 0
}

But I have a question about the depression. On my bench I have the possibility to adjust the speed of the motor to arrive at the famous value of 28 or similar, but when I speed up or speed down the engines, the value of the depression doesn't change. 
Can you explain me how it works this depression value and how can I set my bench at @28 to start the test ? 

Let know if it is not very clear, I will try to explain all better

**Reply by DeeEmm** [reply]
Ok I think I get what you are saying.

At this time I would ignore the 'Adjusted flow rate' as I have not checked / modded the adjusted flow rate code for the orifice style bench. It is possible that it is incorrect. For testing just use the ACFM (Actual) for the present time.

At 28" test pressure, you should see 100cfm flow rate. Or pretty close to that. But you are only seeing 11cfm @35". This obviously does not seem correct. It might possibly be the same error as the baro (decimal point error).

To be able to dial in your reference depression you can either control your motor speed. OR. you can open an air bleed (a valve / hole to let air in). The Depression value on the GUI should show the live reference pressure in all cases. There is no need to do anything else. It is the direct and live reading of the pRef sensor.

So normally you would open all air bleeds. 
Select orifice
Slowly increase vacuum by closing air bleeds (or ramp up motor speed) until you read 28" pRef (reference)
Then the flow value should be correct.

What pRef sensor are you using?

Do you have a secondary method for measuring such as a manometer that can be used to validate / compare?

**Reply by danielerpl** [reply]
Ok, I thinks I got what do you mean 

just a question, if I have @35, in this case I have much motor speed than the requirements ? Because I have 6 motors and I can turno on each one per time as required.
In the other case, is @35 less motor power Thant the required for @28 ? 
I hope I explained well my question 
For the sensors I have the NXP mpxv7002dp

**Reply by danielerpl** [reply]
Secondary method you mean another tool to test the depression ?

**Comment by stefan63** [comment]
Is the "BCD orifice coding pins" nessecary to be used for a pressure drop bench, if so how to use them and why do we need them?

**Reply by DeeEmm** [reply]
BCD pins (Binary Coded Decimal) are to allow the code to detect which orifice is currently in use. At present this functionality is not written. 

When I originally determined the I/O schedule for the shield design, I had to include all possible I/O use cases to be able to fix the hardware design. So there are several references to things like this in the code, but at present they are just placeholders for future functionality / development.

The BCD coding is a way to increase the number permutations that you can achieve from I/O. For example. With straight I/O 3 inputs would allow you to only use 3 orifice plates. But with BCD you can use 9 (3^2). For example...

```

000
001
010
011
100
101
110
111

``` 

Depending on the status of the three inputs, you can have 9 different status'

To implement this you would need to provide three inputs from each orifice. This could be simply three holes drilled in the orifice that are detected by optical switches, or maybe if all orifices are in a rotating disk, switches can detect the position of the disc. 

There are many ways to use this kind of coding. But at present, it is simply enough to manually select which orifice is being used in the GUI.

**Reply by stefan63** [reply]
Thank you, that function must have low priority it is kind of a layzy mans luxury :)

**Reply by DeeEmm** [reply]
Haha yes. I like that. 'lazy mans luxury' lol.

It was a case of thinking. If we could integrate anything we wanted to make the absolute best solution, what would we include.

Ultimately it is part of a fully automated bench where you can put the test piece on the surface, press 'Go' and it automatically maps the flow characteristics across the pre-defined lift range and creates the flow map. This is why it also includes stepper motors - for lift and flow / orifice control.


With regards to your vacuum motors producing too much flow, you can either bleed some air off, or only switch on 1,2,3,4 or 5 motors to get the depression that you need.

Again there is some I/O assigned to control vacuum motors in banks, for this reason, but it is also not yet programmed.

**Reply by danielerpl** [reply]
Ok, nice to this new info about the automated adjustment for the pressure. 
About the bench I will try what did you suggest and I will let you know the result 
Many thanks!! 
I will keep you updated

**Reply by danielerpl** [reply]
Ok, another test it's done. 
- First question about sensors. The Pressure sensor has 2 connection inlet,  I connected to the bottom one and now I can see the right value of the depression I mean.  Can you explain me the difference between  the top and the bottom inlet ? It is correct my connection ? 
In this picture you can see how I connected the tube 
![WhatsApp Image 2024-10-02 at 10 43 53](https://github.com/user-attachments/assets/b7c8ad2e-d86a-425d-909f-26cc13904464)

- Second question about pressure sensor from the bench to the board. In my bench, created by the pts project I have 3 pressure points. The first important 2, before and after the orifice it's ok and connected to the board. At the moment I excluded the valves for flow switching, you can see the valve from the picture, just to make sure I am reading the correct values. Can you explain me how can I connect the static  pressure point from the top of the bench, near the tested piece, to the board ? at the moment it is not connected to the board. From the picture below you can see the static pressure from the bench, not connected to the board 
![WhatsApp Image 2024-10-02 at 12 30 08](https://github.com/user-attachments/assets/c2bd5408-c5c8-4055-bba3-2328b8b90014)
![WhatsApp Image 2024-10-02 at 12 30 35](https://github.com/user-attachments/assets/21a1aaf4-b8c0-4eb7-9c66-8bebef2053c2)

Can you help me to use these valves and make the right connection to the board ? 

- Third point. I made I test and we can now see the depression value change based on the engine flow adjustment.   I started with just one motor, as you can see from the video attached in this message, and I powered up the motor to see the different value of depression. When I have see is, with all 6 motors up and running at the minimum speed I have the @28 with the smallest one orifice previously mentioned in another message, I tried to change it with another one more bigger than the first one, and I am able to spend the motors a little bit up and adjust the depression. 
Do you think my motors hare very powerful for this bench ? I followed the pts schema and specs to make it and chose the motors 

https://github.com/user-attachments/assets/f4f971ed-9688-497e-a6e4-2ac3b9aecaf8


Many thanks to all your support! 

Now I think I am ready to make a first test with a real piece on top of the bench! I am pretty sure we don't need to make more changes at the code for the orifice section! Congrats again for the great job!!

**Reply by DeeEmm** [reply]
The differential pressure sensors are the same as a regular sensor but have a barb each side of the sensing element (the regular sensor has one side of the sensing element open to atmosphere). The two barbs allow you to measure the difference in pressure between two locations.

You can also use a regular sensor with one barb to measure these same points provided that the sensor type can measure vacuum. However such a sensor would not be able to measure exhaust (reverse) flow.

Generally the differential pressure sensors can measure both vacuum and pressure (-/+) which allows them to be used to measure both regular (vacuum) flow and exhaust (pressure) flow.

For the reference pressure we have one hose connected to the orifice plenum and the other hose to atmosphere. This measures the depression that we are providing to the orifice. 

As long as we adjust to the same depression that the orifice was calibrated at, we will be able to use the orifice's calibration data to work out how much flow we are seeing at the test piece. 

For the differential pressure, we have one hose connected to the top plenum (in between orifice plate and measured object) and the other to atmosphere. The name differential pressure may be a little misleading here. But it is used to calculate the differential between the two plenums. If the relationship is 50%, then the resultant flow figure will be 50% of the orifice's calibrated flow value.

Determining if your vacuum motors are 'too large'. Fit the largest plenum you have, then without a test piece see if you can achieve the calibration depression. 

To determine the largest plenum size you need, just consider the highest flowing object you will test.

Most home built benches struggle to achieve a full 28" with the largest orifice, so if you have excess capacity, this is a good thing. As mentioned before, it is fairly easy to regulate the reference depression down. it is much harder to increase it.

The other end of the scale is the lowest flow you will measure, i.e. the smallest orifice. If you can also regulate the depression with a single motor, then you should be able to measure everything in between.

Some form of air bleed may be necessary to fine tune. But I would be interested to see if you can regulate the motor speed to achieve accuracy with both small and large orifices.

Let me know how you get on.

The next thing that we will need to consider is temperature differential. This is especially pertinent if your vacuum motors are in between the test piece and orifice plate. This will also require an additional temperature sensor.

**Reply by danielerpl** [reply]
Ok perfect explanation, il make this test to make the regulation with each orifice size. I already  tried the smallest and the bigger and I am able to adjust the bench to @28 without any problem! I just need to turn on or of some motors but it is easy. 

I just need to understand well one point. The static pressure point that I have on my bench, where have to be connected to the board ?

**Reply by DeeEmm** [reply]
When you say 'static pressure' is this a specific feature of your bench?

**Reply by danielerpl** [reply]
![WhatsApp Image 2024-10-02 at 12 30 08](https://github.com/user-attachments/assets/fcab83dd-8ad7-4e06-835b-0cb94bb67063)
![WhatsApp Image 2024-10-02 at 12 30 35](https://github.com/user-attachments/assets/3a65cecc-09ef-49e0-893e-b301057d6128)


I made in red a mark on the picture. Do we need this for our board ? 
Also, how can I use these valves ? it is a feature to switch the bench in vacuum or exhaust flow test

**Reply by DeeEmm** [reply]
I am not familiar with the PTS flow bench design, so it is hard for me to comment on the connections in the drawing as I have no idea of the air path or the function of each plenum. it should be possible to connect the bench as shown in the diagram but may require some change to the maths in the code, which is not a big issue.

I suspect that I may need to differentiate between a few different bench designs as there is more than one way to connect & measure. 

Flick me a copy of the documentation that you have and I will take a look - deeemm[at]deeemm[dot]com

**Comment by stefan63** [comment]
Isn't static pressure the same as reference pressure = the point where we
messure the dialed in 28" standard pressure mo flow benches work at!?st

Den tors 3 okt. 2024 kl 09:34 skrev Mick Percy ***@***.***>:

**Reply by DeeEmm** [reply]
Yes it should be.

**Comment by danielerpl** [comment]
@DeeEmm before a make some others test with this new code. Please can you explain step by step how to calibrate the bench with the officers  ?
Also there are some test buttons in the configuration  page. I tried to get some info from the wiki page, but I have difficult to understand  how it works. To avoid some mistake, please can you explain me these buttons how they work and how to calibrate properly the bench ? 

many thanks

**Comment by black-top** [comment]
Hello, @danielerpl, can you take a picture of your PTS plates? And maybe you can measure the angle? It might be 45 degrees for all plates.  I'm planning to make a set to have at least a rough comparison for now, because a PTS set with shipping and all duties and taxes might be over 400eu, so I'm planning to buy it at the feature, when the main issues will be sorted with my flow bench setup.

**Reply by danielerpl** [reply]
I will make a picture for you. In the meantime I leave you the website of the orifice I got. I am pretty sure it is 45 degrees 
http://www.flowbenchtech.com/orifice_plates.html

**Reply by danielerpl** [reply]
@black-top scrolling in all this comments, you can see the picture of my plates in this section. There is a comment with the orifice I used to make some test

**Comment by DeeEmm** [comment]
I've made sharp edge orifice plates in the past and recall that 45deg is fine. 

I had originally made the plates with a much sharper edge but was challenged on the angle. After checking the maths there is no advantage to a sharper angle than 45 degrees.

45 degree's also makes them a LOT easier to make.

There is a formula you can use to determine the flow amount based on orifice size. I will see if I can dig it up

**Comment by DeeEmm** [comment]
![IMG_4781](https://github.com/user-attachments/assets/4efe4992-24d6-4bbc-b2ff-9f33044f3ee1)

IIRC this was 30 degrees, but 45 is actually fine. The 30 degree was a PITA to achieve

![IMG_4772](https://github.com/user-attachments/assets/b0acd3f0-b3c0-420b-ae68-3fe0ff7fdcb2)

45 edge allows you to use a 45 degree tool which makes it super easy.

**Comment by black-top** [comment]
Yes, 45 degree will be easier to make. I think I will make it from 4mm steel plates.
 It would be good to get PTS plates diameters and expected flow numbers, it will give ability at least for rough comparison.

**Reply by DeeEmm** [reply]
I also found with the aluminium at 30 degree's the edge was so sharp that it easily deformed, even just measuring it with the callipers. I think that steel would be more appropriate and also has more temperature stability. 

I also purchased some calibrated orifice plates from ebay. There were a bunch of them for sale so I grabbed a few different sizes. Those were stainless steel and very nicely made.

![PXL_20230519_021603892](https://github.com/user-attachments/assets/b51773f4-179a-4573-9c97-df87c5e604c3)

![PXL_20230519_021616542](https://github.com/user-attachments/assets/2786ac98-eb34-42b2-9db2-c1644cc8870b)

**Comment by DeeEmm** [comment]
I recall that some back we were discussing using water to physically measure the flow rate of an orifice. The orifice was placed on the outlet of a large reservoir, and then the discharge collected over a set time period. The volume could then be converted to the equivalent volume of air using environmental markers (temp / baro / etc).

The basic idea was how can we (anyone) validate the ACTUAL flow of an orifice using simple tools.

Here's that spreadsheet I mentioned for calculating orifice sizes / flow rates. You will need to change the formatting as the lower output is truncated

[Flowbench Calculations.xls](https://github.com/user-attachments/files/17612518/Flowbench.Calculations.xls)

**Reply by black-top** [reply]
Yes, I remember that, but it would require a lot of time and work to make it work properly, so for now, I think the "copying PTS sizes" will be a quicker and good enough solution for now.

**Comment by danielerpl** [comment]
@DeeEmm I made some test withe the last code ( [5663714] 2 days ago ) 
Upload from bin file it works amazing!! 
Upload index.html with the new flow it's perfect! 

Reading values I have some problems : 

First I wanna show you, how I connected the sensors to the board

About the **PDiff** and **PReff** should be ok because I followed the previous instructions. I just need a confirmation for the connection to the + or - of the sensor 
As you can see, the PDiff sensor is connected to the point before the orifice and PReff is connected to the point after the orifice. 
               
            Air Flow
______|     ⥥    |_______  = Testing peace 

---------------- O    -----> PDiff 

--------0 ----------  = ORIFICE 

---------------- O    -----> PRef  

The point from the top of the bench should be the static pressure from the bench. 
I hope it's connected well 

![IMG_8239](https://github.com/user-attachments/assets/
![IMG_8238](https://github.com/user-attachments/assets/523685c9-248d-45af-99de-0f7e7b31f23d)
c799fa8b-788e-4442-a4d1-595ecc5d9921)
![IMG_8237](https://github.com/user-attachments/assets/9f4e94d2-33d3-4102-ba65-97a42f959153)

Now the problem is, the depression value it is stack! is not changing. What I am expecting is the possibility to arrive at the desired amount of @28 of h20 but when I speed up the motors nothing is changing. 

I am following the procedure for the configuration for the leak test but I am not receiving the expected result. 
What I am doing is : 
 - Close the top of the bench and any other possible hole 
 - Turn on the bench 
 - Press the leak button to save the value 
 The value it's properly set to the Leak Test Offset

Now I am putting the CD on top of the bench to make a calibration test for the flow calibration offset but I am not reading the expected value. 
So, the first thing is.... with the new dashboard, where I have to read the @28 or the value I would like to set to make my test ? when I speed up or down the motors, I am not able to red this value like the previous dashboard. My depression value never change from 3.6x 

Please help me because I am very confused with this new dashboard  and the bench is not going in the right direction with these values. 
Another point is, if I copy the leak value to the baseline value like described in the procedure, in the dashboard I will have negative values. 
Where I making the huge mistake for the configuration ? 
The dashboard values from the picture it's about the test with the CD

![IMG_8241](https://github.com/user-attachments/assets/796f52fa-f204-4b68-b1cf-97c881700fa8)

<img width="1680" alt="Screenshot 2024-11-06 at 14 40 32" src="h
<img width="906" alt="Screenshot 2024-11-06 at 16 43 50" src="https://github.com/user-attachments/assets/a04a0d3e-7507-470c-a44b-41e911a4f1e6">
ttps://github.com/user-attachments/assets/d81bbe65-bf7e-4b13-a635-4f56c7c4af93">
<img width="872" alt="Screenshot 2024-11-06 at 16 43 39" src="https://github.com/user-attachments/assets/a2d8f4b6-c29d-4aca-8335-ac04101db6a3">

**Reply by KirikauKiwi** [reply]
Are the Static pickup and Top DP pickup ports in the same chamber of the flow bench?

If they are you will need to disconnect the purple tube that is connected between Static pickup to both PRef and PDiff on the DIYFB shield. 
PRef should be connected to the Top DP Pickup and PDiff should be connected to the Bottom DP pickup. 

Make sure to plug the Static pickup, or you will have an air leak. It is not needed for the DIYFB shield.

**Reply by KirikauKiwi** [reply]
Looking at the drawings above for the construction of the PTS flow bench.

Static Pickup = Top DP pickup = PRef, they are all measuring the same pressure.
Bottom DP pickup = PDiff

Comparing the PTS Manometer to the DIYFB shield the way they establish the new pressure is done differently.
**DTS Manometer**
PT1 is measuring the pressure above the internal orifice plate.
PT2 is measuring the delta pressure e.g the pressure drop across the plate, Delta P = Top DP Pickup - Bottom DP Pickup.

Example.
So you adjust the flow bench to 28 inH2O at PT1 and then PT2 would measure 6 inH2O
New Pressure = PT1 - PT2
New Pressure = 28 - 6
New Pressure = 22 in H2O
Here the Delta P = PT2 this is actually measured by the manometer/pressure sensor and is needed to calculate the new pressure


**DIYFB Shield**
PRef is measuring the pressure above the orifice plate.
PDiff is measuring the pressure below the orifice plate.

Example
Adjust the flow bench to 28 inH2O at PRef and PDiff would measure 22 inH20
Old Pressure = PRef
New Pressure = PDiff

This is the ratio metric formula then used to calculate the new CFM value.

CFM₂ = CFM₁ × √(P₁ / P₂)

Where:

CFM₁ = The airflow (in CFM) at the old pressure (P₁)
CFM₂ = The airflow (in CFM) at the new pressure (P₂)
P₁ = The old pressure (in inches of water column)
P₂ = The new pressure (in inches of water column)

The PTS flow bench has to calculate the new pressure, where the DIYFB measures the new pressure directly. @DeeEmm please chime in if I have got this wrong.

**Reply by DeeEmm** [reply]
@danielerpl  I think this may also possibly be related to the issue that we were discussing about zeroing the pressure sensor values down. There is an error in the code that is preventing negative values.

At the moment I am without an internet connection and typing on my phone / tablet is just too painful. I will take a look at this once the connection is fixed.

**Reply by danielerpl** [reply]
@KirikauKiwi amazing explanation very helpful. 
Yes static pickup, top pickup and bottom are in the same chamber. I was thinking to do this operation but I was waiting for a confirmation that I have now 

About the famous value of 28 inH2O, where I will see this value in the DIYFB Dashboard ? 
Do I have the two pressure on the dashboard ? 

I think I already understood this, but your confirmation will be also helpful 

Sorry if I am in trouble with the first power on, but this is my first bench :D

**Reply by KirikauKiwi** [reply]
@danielerpl The famous 28 inH20 should be displayed under "Depression"

"Depression" and "Diff Pressure" are displayed in the same place on the GUI, click on it and it will toggle between them both.
<img width="743" alt="Screenshot 2024-11-07 at 14 04 34" src="https://github.com/user-attachments/assets/64311f4e-a07b-433c-ad22-50457e95aa9d">

You should have a negative value being displayed when the bench is running, if you have a positive value displayed, switch your purple tube to the other port on the PRef and PDiff sensors e.g. if they are close to the PCB move them to the port away from the PCB.

**Reply by danielerpl** [reply]
Ok perfect, I will try in the next days and I will let you know.  I am pretty sure my mistake is the connection of the static pressure to the sensors

**Reply by danielerpl** [reply]
Final test done!  everything it's working perfectly! 
Only one thing,  I am not able to have the depression in negative value. I tried to change the pressure point connection to the board but in some cases is not working the final ACFM value and  in other is not working the depressione. I think it's not a problem if I have the positive value. Apparently it's working fine. I made a test with a pice original and after the preparation and the result ACFM flow it's appear true

**Reply by DeeEmm** [reply]
What version do you have installed? 

There were some changes to the code to allow this, but they may not be published to the main branch yet.
